projects/packages/doc-parser/src/class-doc-parser.php

```diff
@@ -0,0 +1,92 @@
+<?php
+/**
+ * Package description here
+ *
+ * @package automattic/jetpack-doc-parser
+ */
+
+namespace Automattic\Jetpack;
+
+/**
+ * Converts PHPDoc markup into a template ready for import to a WordPress blog.
+ */
+class Doc_Parser {
+
+	const PACKAGE_VERSION = '0.1.0-alpha';
+
+	/**
+	 * Generate a JSON file containing the PHPDoc markup, and save to filesystem.
+	 *
+	 * @param Array $args this function takes a path as its argument,
+	 * as well as optionally an output file name.
+	 */
+	public function generate( $args ) {
+		list( $directories, $output_file ) = $args;
+
+		if ( empty( $output_file ) ) {
+			$output_file = 'phpdoc.json';
+		}
+
+		$json = array();
+		foreach ( $directories as $directory ) {
+			$directory = realpath( $directory );
+			echo PHP_EOL;
+
+			// Get data from the PHPDoc
+			$json = array(
+				...$json,
+				$this->get_phpdoc_data( $directory ),
+			);
```

Comment:
To fix the "PHP lint" jobs on this line, we'll have to add `SKIPS` for appropriate PHP versions in `tools/parallel-lint.sh`.

I wonder if I can make that read from composer.json though...

OTOH, you could avoid all the trouble by just using the normal PHP idiom like this instead:
```suggestion
			$json[] = $this->get_phpdoc_data( $directory );
```

Reply:
> I wonder if I can make that read from composer.json though...

#34497